src/SAMLTrace.js

```diff
@@ -405,6 +422,101 @@ SAMLTrace.RequestItem.prototype = {
     }
     target.appendChild(doc.createTextNode(samlFormatted));
   },
+  
+  'showSummary' : function(target) {
+    var doc = target.ownerDocument;
+    var samlSummary = "";
+    var parser  = new DOMParser();
+    var xmldoc  = parser.parseFromString(this.request.saml,"text/xml");
+
+    /* Check for AuthnRequest */
+    var AuthnRequest = xmldoc.getElementsByTagNameNS('*','AuthnRequest');
+    if (AuthnRequest.length>0) {
+        samlSummary += '\nAuthnRequest: \n';
+        samlSummary += this.summaryAddAttVal(AuthnRequest[0],'Destination');
+        samlSummary += this.summaryAddAttVal(AuthnRequest[0],'ForceAuthn');
+        samlSummary += this.summaryAddAttVal(AuthnRequest[0],'AssertionConsumerServiceURL');
+    }
+    
+    /* Check for Issuer*/
+    var Issuer = xmldoc.getElementsByTagNameNS('*','Issuer');
+    if (Issuer.length>0) { 
+        samlSummary += this.summaryAddAttText(Issuer[0],'Issuer');
+    }
+
+    /* Check for Subject */
+    var Subject = xmldoc.getElementsByTagNameNS('*','Subject');
+    if (Subject.length>0) {
+        samlSummary += this.summaryAddAttText(Subject[0],'Subject');
+    }
+
+    /* Check for NameID */
+    var NameID = xmldoc.getElementsByTagNameNS('*','NameID');
+    if (NameID.length>0) {
+        samlSummary += this.summaryAddAttText(NameID[0],'NameID');
+    }
+    /* Check for AttributeStatement */
+    var AttributeStatement = xmldoc.getElementsByTagNameNS('*','AttributeStatement');
+    if (AttributeStatement.length>0) {
+        var AttributeStatementChilds = AttributeStatement[0].childNodes;
+        if (AttributeStatementChilds.length>0) { 
+            samlSummary += "\nAttributeStatement:\n";}
+        for (i = 0; i < AttributeStatementChilds.length; i++) {
+            var attribute = AttributeStatementChilds[i];
+            var attributeName = attribute.getAttribute('Name');
+            var attributeValues = [];
+            for (j = 0; j<attribute.childNodes.length; j++) {
+                    samlSummary += this.summaryAddAttText(attribute.childNodes[j],attributeName,' * ',this.longPadding);
+            }
+            
+        }
+    }
+    
+    /* Check for LogoutRequest */
+    var LogoutRequest = xmldoc.getElementsByTagNameNS('*','LogoutRequest');
+    if (LogoutRequest.length>0) {
+        samlSummary += '\nLogoutRequest: \n';
+        samlSummary += this.summaryAddAttVal(AuthnRequest[0],'Destination');
+    }
+    
+
+    target.appendChild(doc.createTextNode(samlSummary));
+  },
+  
```

Comment:
The functions `summaryAddAttVal` and `summaryAddAttText` are very similar. And I think the try/catch isn't really needed (at least I couldn't cause an exception while debugging). So to reduce code duplication you could adjust the two functions like this:

```js
'summaryAddAttVal' : function(node,attrName,decorator='',padLen=this.shortPadding) {
  let attr = node.attributes[attrName];
  return this.summaryAdd(attr, description, decorator, padLen);
},

'summaryAddAttText' : function(value,description,decorator='',padLen=this.shortPadding) {
  let attr = value.textContent.trim();
  return this.summaryAdd(attr, description, decorator, padLen);
},

'summaryAdd' : function(value,description,decorator,padLen) {
  if (value.length>0) {
    let s="";
    s += (decorator+description).padEnd(padLen," ");
    s += "= "+value;
    s += "\n";
    return s;
  }
},
```